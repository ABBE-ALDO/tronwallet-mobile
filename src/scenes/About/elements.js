import styled, { css } from 'styled-components'
import { Colors } from '../../components/DesignSystem'

const BaseText = css`
  color: ${Colors.secondaryText};
  padding: 20px;
  align-self: center;
`
export const Description = styled.Text`
  font-family: Rubik-Light;
  font-size: 16px;
  line-height: 20px;

  color: white;
`
export const VersionText = styled.Text`
  font-family: Rubik-Regular;
  font-size: 14px;
  letter-spacing: 0.6;
  ${BaseText}
`
export const SectionTitle = styled.Text`
  margin-top: 5px;
  font-family: Rubik-Bold;
  font-size: 14px;
  ${BaseText}
`
export const PayPartner = styled.Image`
  width: 120px;
  height: 50px;
  resize-mode: contain;
`
export const Getty = styled.Image`
  width: 150px;
  height: 50px;
  resize-mode: contain;
`
const PaddingVertical = css`
  padding-vertical: 10px;
`
export const DescriptionWrapper = styled.View`
  ${PaddingVertical}
`
export const Partners = styled.View`
  ${PaddingVertical}
`
export const TutorialWrapper = styled.View`
  padding: 20px;
  background-color: ${Colors.secondaryText};
  border-radius: 4px;
`
export const TutorialText = styled.Text`
  font-family: Rubik-Bold;
  font-size: 14px;
  color: white;
  text-align: center;
`
