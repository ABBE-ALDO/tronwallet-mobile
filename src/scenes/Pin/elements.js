import React from 'react'
import styled from 'styled-components'
import Ionicons from 'react-native-vector-icons/Ionicons'
import { transparentize } from 'polished'

import { Colors, Spacing } from '../../components/DesignSystem'

export const Label = styled.Text`
  font-family: Rubik-Medium;
  font-size: 11px;
  color: #66688F;
  letter-spacing: 0.6;
  line-height: 12px;
`

export const Text = styled.Text`
  font-family: Rubik-Medium;
  font-size: 16px;
  color: #FFF;
  line-height: 18px;
`

export const KeyPad = styled.View`
  flex-direction: row;
  flex-wrap: wrap;
  height: 242px;
  margin: 10px;
`

export const KeyWrapper = styled.TouchableOpacity`
  height: 52px;
  align-items: center;
  justify-content: center;
  flex-basis: 30%;
  flex-grow: 1;
  flex-shrink: 0;
  border-radius: 25px;
  border-width:${props => props.noBorder ? 0 : 1}px;
  border-color:${Colors.secondaryText};
  margin: 4px;
`

export const KeyText = styled.Text`
  font-family: Rubik-Regular;
  font-size: 16px;
  line-height: 24px;
  color:${Colors.secondaryText};
`

export const Key = ({ children, noBorder, onPress }) => (
  <KeyWrapper noBorder={noBorder} onPress={() => onPress(children)} >
    <KeyText>
      {children}
    </KeyText>
  </KeyWrapper>
)

export const Wrapper = styled.View`
  width: 36px;
  height: 36px;
  border-radius: 18px;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.color};
`

Wrapper.defaultProps = {
  color: 'transparent'
}

export const Circle = styled.View`
  background-color: ${props => transparentize((1 - props.opacityAmount), props.color)};
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  border-radius: ${props => props.size / 2}px;
  align-items: center;
  justify-content: center;
  z-index: 1;
`

Circle.defaultProps = {
  size: 12,
  opacityAmount: 1,
  color: '#2E2F47'
}

export const PinDigit = ({ digit, currentState }) => {
  if (digit < currentState.length) {
    return (
      <Wrapper>
        <Circle color='#9b9cb7' />
      </Wrapper>
    )
  }
  if (digit > currentState.length) {
    return (
      <Wrapper>
        <Circle />
      </Wrapper>
    )
  }
  return (
    <Wrapper>
      <Circle />
    </Wrapper>
  )
}

const GoBackBtn = styled.TouchableOpacity`
  position: absolute;
  top: 10;
  left: 8;
  padding: ${Spacing.medium}px;
  z-index: 1;
`

export const GoBackButton = props => (
  <GoBackBtn {...props} >
    <Ionicons name='ios-arrow-round-back' size={32} color='#FFF' />
  </GoBackBtn>
)

export const BiometricButton = styled.TouchableOpacity`
  padding: ${Spacing.medium}px;
`
