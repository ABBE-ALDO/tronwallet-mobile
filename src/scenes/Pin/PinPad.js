import React, { PureComponent } from 'react'
import { Vibration, Platform } from 'react-native'
import Ionicons from 'react-native-vector-icons/Ionicons'
import * as Animatable from 'react-native-animatable'
import PropTypes from 'prop-types'

import { PinDigit, KeyPad, Key, PinWrapper } from './elements'
import * as Utils from '../../components/Utils'
import { Colors } from '../../components/DesignSystem'

class PinPad extends PureComponent {
  static propTypes = {
    onComplete: PropTypes.func
  }

  static defaultProps = {
    onComplete: () => null
  }

  state = {
    pin: []
  }

  componentDidUpdate (_, prevState) {
    if (this.state.pin.length > prevState.pin.length && this.state.pin.length === 6) {
      setTimeout(() => {
        this.props.onComplete(this.state.pin.join(''), this.state.pin)
      }, 500)
    }
  }

  reset = () => this._handleClear()

  shake = () => {
    Vibration.vibrate(500)
    this.pinView.shake(1000)
      .then(endState => {
        if (endState.finished) {
          Vibration.cancel()
        }
      })
      .catch(() => {
        Vibration.cancel()
      })
  }

  _checkPin = () => {
    if (this.state.pin.length === 6) {
      this.props.onComplete(this.state.pin.join(''), this.state.pin)
    }
  }

  _handleKeyInput = digit => {
    const { pin } = this.state
    Vibration.vibrate(30)
    if (pin.length < 6) {
      this.setState({
        pin: pin.concat(parseInt(digit))
      }, () => {
        if (Platform.OS === 'ios') {
          Vibration.cancel()
        }
      })
    }
  }

  _handleDelete = () => {
    const { pin } = this.state

    this.setState({
      pin: pin.slice(0, -1)
    })
  }

  _handleClear = () => {
    this.setState({ pin: [] })
  }

  render () {
    const { pin } = this.state
    return (
      <Utils.View marginBottom={10}>
        <PinWrapper>
          <Animatable.View ref={ref => { this.pinView = ref }} >
            <Utils.Row align='center' justify='space-between' width={264}>
              <PinDigit digit={0} currentState={pin} />
              <PinDigit digit={1} currentState={pin} />
              <PinDigit digit={2} currentState={pin} />
              <PinDigit digit={3} currentState={pin} />
              <PinDigit digit={4} currentState={pin} />
              <PinDigit digit={5} currentState={pin} />
            </Utils.Row>
          </Animatable.View>
        </PinWrapper>

        <KeyPad>
          <Key onPress={this._handleKeyInput}>1</Key>
          <Key onPress={this._handleKeyInput}>2</Key>
          <Key onPress={this._handleKeyInput}>3</Key>
          <Key onPress={this._handleKeyInput}>4</Key>
          <Key onPress={this._handleKeyInput}>5</Key>
          <Key onPress={this._handleKeyInput}>6</Key>
          <Key onPress={this._handleKeyInput}>7</Key>
          <Key onPress={this._handleKeyInput}>8</Key>
          <Key onPress={this._handleKeyInput}>9</Key>
          <Key disabled noBorder />
          <Key onPress={this._handleKeyInput}>0</Key>
          <Key
            onPress={this._handleDelete}
            onLongPress={this._handleClear}
            pointerEvents={pin.length === 0 ? 'none' : 'auto'}
          >
            <Ionicons name='ios-backspace' size={24} color={Colors.secondaryText} />
          </Key>
        </KeyPad>
      </Utils.View>
    )
  }
}

export default PinPad
