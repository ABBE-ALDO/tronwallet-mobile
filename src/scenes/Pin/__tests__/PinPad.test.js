import React from 'react'
import 'react-native'
import PinPad from './../PinPad'
import { Key } from './../elements'

import { shallow } from 'enzyme'
import 'jest-styled-components'

jest.mock('react-native-fabric')
jest.mock('react-native-mixpanel')
jest.mock('../../../utils/i18n')
jest.mock('Vibration', () => ({
  vibrate: jest.fn(),
  cancel: jest.fn()
}))

const getRandom = (limit = 10) => Math.floor((Math.random() * limit))

describe('Pin Scene > PinPad', () => {
  const defaultProps = {
    onComplete: jest.fn()
  }
  const setup = (propOverrides = {}) => {
    const setupProps = { ...defaultProps, ...propOverrides }

    const wrapper = shallow(
      <PinPad {...setupProps} />
    )

    return {
      wrapper
    }
  }

  describe('Key button press', () => {
    test('should call _handleKeyInput when the key-button is clicked', () => {
      const { wrapper } = setup()
      const instance = wrapper.instance()
      const spy = jest.spyOn(instance, '_handleKeyInput')
      instance.forceUpdate()
      const btnKey1 = wrapper.find(Key).first()
      btnKey1.props().onPress(1)
      expect(spy).toHaveBeenCalled()
      spy.mockRestore()
    })

    test('should update the pin into state when some key-button is clicked', () => {
      const { wrapper } = setup()
      const randomKey = getRandom(9)
      const btnKey = wrapper.find(Key).at(randomKey)
      const btnValue = parseInt(btnKey.props().children)
      btnKey.props().onPress(btnValue)
      expect(wrapper.state('pin')).toEqual([btnValue])
    })

    test('should delete last value when backspace-button is clicked', () => {
      const { wrapper } = setup()
      wrapper.setState({ pin: [1, 1, 1] })
      const btnBackspace = wrapper.find(Key).at(11)
      const btnKey1 = wrapper.find(Key).first()
      btnKey1.props().onPress(1)
      btnBackspace.props().onPress()
      expect(wrapper.state('pin')).toEqual([1, 1, 1])
    })

    test('should clean value when backspace-button is long pressed', () => {
      const { wrapper } = setup()
      wrapper.setState({ pin: [1, 1, 1, 1, 1, 1] })
      const btnBackspace = wrapper.find(Key).at(11)
      btnBackspace.props().onLongPress()
      expect(wrapper.state('pin')).toEqual([])
    })
  })

  describe('On input complete', () => {
    test('should call #onComplete when state lenght is 6', () => {
      const { wrapper } = setup()
      wrapper.setState({ pin: [1, 1, 1, 1, 1] })
      const instance = wrapper.instance()
      const spy = jest.spyOn(instance, '_handleKeyInput')
      instance.forceUpdate()
      const btnKey1 = wrapper.find(Key).first()
      btnKey1.props().onPress(1)
      expect(wrapper.state('pin').length).toEqual(6)
      expect(spy).toHaveBeenCalled()
      spy.mockRestore()
    })

    test('test submit value when #onComplete is called', () => {
      const onComplete = jest.fn(pin => pin)
      const { wrapper } = setup({ onComplete })
      let inputValue = []
      for (let cont = 0; cont < 6; cont++) {
        const key = getRandom(9)
        const btnKey = wrapper.find(Key).at(key)
        btnKey.props().onPress(
          btnKey.props().children
        )
        inputValue.push(
          btnKey.props().children
        )
      }

      expect(wrapper.state('pin').length).toEqual(6)
      expect(onComplete.mock.results[0].value).toEqual(inputValue.join(''))
    })
  })

  test('Should match snapshot', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })
})
