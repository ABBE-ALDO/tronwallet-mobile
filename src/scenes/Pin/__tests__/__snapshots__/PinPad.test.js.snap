// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pin Scene > PinPad Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PinPad
    onComplete={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "align": "stretch",
      "children": Array [
        <Styled(View)
          align="center"
          height={180}
          justify="center"
        >
          <withAnimatable(View)
            delay={0}
            direction="normal"
            iterationCount={1}
            iterationDelay={0}
            onAnimationBegin={[Function]}
            onAnimationEnd={[Function]}
            onTransitionBegin={[Function]}
            onTransitionEnd={[Function]}
            useNativeDriver={false}
          >
            <Styled(View)
              align="center"
              justify="space-between"
              width={264}
            >
              <PinDigit
                currentState={Array []}
                digit={0}
              />
              <PinDigit
                currentState={Array []}
                digit={1}
              />
              <PinDigit
                currentState={Array []}
                digit={2}
              />
              <PinDigit
                currentState={Array []}
                digit={3}
              />
              <PinDigit
                currentState={Array []}
                digit={4}
              />
              <PinDigit
                currentState={Array []}
                digit={5}
              />
            </Styled(View)>
          </withAnimatable(View)>
        </Styled(View)>,
        <Styled(View)>
          <Key
            onPress={[Function]}
          >
            1
          </Key>
          <Key
            onPress={[Function]}
          >
            2
          </Key>
          <Key
            onPress={[Function]}
          >
            3
          </Key>
          <Key
            onPress={[Function]}
          >
            4
          </Key>
          <Key
            onPress={[Function]}
          >
            5
          </Key>
          <Key
            onPress={[Function]}
          >
            6
          </Key>
          <Key
            onPress={[Function]}
          >
            7
          </Key>
          <Key
            onPress={[Function]}
          >
            8
          </Key>
          <Key
            onPress={[Function]}
          >
            9
          </Key>
          <Key
            disabled={true}
            noBorder={true}
          />
          <Key
            onPress={[Function]}
          >
            0
          </Key>
          <Key
            onLongPress={[Function]}
            onPress={[Function]}
            pointerEvents="none"
          >
            <Icon
              allowFontScaling={false}
              color="#66688F"
              name="ios-backspace"
              size={24}
            />
          </Key>
        </Styled(View)>,
      ],
      "justify": "flex-start",
      "marginBottom": 40,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "children": <withAnimatable(View)
            delay={0}
            direction="normal"
            iterationCount={1}
            iterationDelay={0}
            onAnimationBegin={[Function]}
            onAnimationEnd={[Function]}
            onTransitionBegin={[Function]}
            onTransitionEnd={[Function]}
            useNativeDriver={false}
          >
            <Styled(View)
              align="center"
              justify="space-between"
              width={264}
            >
              <PinDigit
                currentState={Array []}
                digit={0}
              />
              <PinDigit
                currentState={Array []}
                digit={1}
              />
              <PinDigit
                currentState={Array []}
                digit={2}
              />
              <PinDigit
                currentState={Array []}
                digit={3}
              />
              <PinDigit
                currentState={Array []}
                digit={4}
              />
              <PinDigit
                currentState={Array []}
                digit={5}
              />
            </Styled(View)>
          </withAnimatable(View)>,
          "height": 180,
          "justify": "center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animation": undefined,
            "children": <Styled(View)
              align="center"
              justify="space-between"
              width={264}
            >
              <PinDigit
                currentState={Array []}
                digit={0}
              />
              <PinDigit
                currentState={Array []}
                digit={1}
              />
              <PinDigit
                currentState={Array []}
                digit={2}
              />
              <PinDigit
                currentState={Array []}
                digit={3}
              />
              <PinDigit
                currentState={Array []}
                digit={4}
              />
              <PinDigit
                currentState={Array []}
                digit={5}
              />
            </Styled(View)>,
            "delay": 0,
            "direction": "normal",
            "duration": undefined,
            "easing": undefined,
            "iterationCount": 1,
            "iterationDelay": 0,
            "onAnimationBegin": [Function],
            "onAnimationEnd": [Function],
            "onTransitionBegin": [Function],
            "onTransitionEnd": [Function],
            "style": undefined,
            "transition": undefined,
            "useNativeDriver": false,
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "center",
              "children": Array [
                <PinDigit
                  currentState={Array []}
                  digit={0}
                />,
                <PinDigit
                  currentState={Array []}
                  digit={1}
                />,
                <PinDigit
                  currentState={Array []}
                  digit={2}
                />,
                <PinDigit
                  currentState={Array []}
                  digit={3}
                />,
                <PinDigit
                  currentState={Array []}
                  digit={4}
                />,
                <PinDigit
                  currentState={Array []}
                  digit={5}
                />,
              ],
              "justify": "space-between",
              "width": 264,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currentState": Array [],
                  "digit": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currentState": Array [],
                  "digit": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currentState": Array [],
                  "digit": 2,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currentState": Array [],
                  "digit": 3,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currentState": Array [],
                  "digit": 4,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currentState": Array [],
                  "digit": 5,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Key
              onPress={[Function]}
            >
              1
            </Key>,
            <Key
              onPress={[Function]}
            >
              2
            </Key>,
            <Key
              onPress={[Function]}
            >
              3
            </Key>,
            <Key
              onPress={[Function]}
            >
              4
            </Key>,
            <Key
              onPress={[Function]}
            >
              5
            </Key>,
            <Key
              onPress={[Function]}
            >
              6
            </Key>,
            <Key
              onPress={[Function]}
            >
              7
            </Key>,
            <Key
              onPress={[Function]}
            >
              8
            </Key>,
            <Key
              onPress={[Function]}
            >
              9
            </Key>,
            <Key
              disabled={true}
              noBorder={true}
            />,
            <Key
              onPress={[Function]}
            >
              0
            </Key>,
            <Key
              onLongPress={[Function]}
              onPress={[Function]}
              pointerEvents="none"
            >
              <Icon
                allowFontScaling={false}
                color="#66688F"
                name="ios-backspace"
                size={24}
              />
            </Key>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "1",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "1",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "2",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "2",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "3",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "3",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "4",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "4",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "5",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "5",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "6",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "6",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "7",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "7",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "8",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "8",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "9",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "9",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": true,
              "noBorder": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "0",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "0",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Icon
                allowFontScaling={false}
                color="#66688F"
                name="ios-backspace"
                size={24}
              />,
              "onLongPress": [Function],
              "onPress": [Function],
              "pointerEvents": "none",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowFontScaling": false,
                "color": "#66688F",
                "name": "ios-backspace",
                "size": 24,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "stretch",
        "children": Array [
          <Styled(View)
            align="center"
            height={180}
            justify="center"
          >
            <withAnimatable(View)
              delay={0}
              direction="normal"
              iterationCount={1}
              iterationDelay={0}
              onAnimationBegin={[Function]}
              onAnimationEnd={[Function]}
              onTransitionBegin={[Function]}
              onTransitionEnd={[Function]}
              useNativeDriver={false}
            >
              <Styled(View)
                align="center"
                justify="space-between"
                width={264}
              >
                <PinDigit
                  currentState={Array []}
                  digit={0}
                />
                <PinDigit
                  currentState={Array []}
                  digit={1}
                />
                <PinDigit
                  currentState={Array []}
                  digit={2}
                />
                <PinDigit
                  currentState={Array []}
                  digit={3}
                />
                <PinDigit
                  currentState={Array []}
                  digit={4}
                />
                <PinDigit
                  currentState={Array []}
                  digit={5}
                />
              </Styled(View)>
            </withAnimatable(View)>
          </Styled(View)>,
          <Styled(View)>
            <Key
              onPress={[Function]}
            >
              1
            </Key>
            <Key
              onPress={[Function]}
            >
              2
            </Key>
            <Key
              onPress={[Function]}
            >
              3
            </Key>
            <Key
              onPress={[Function]}
            >
              4
            </Key>
            <Key
              onPress={[Function]}
            >
              5
            </Key>
            <Key
              onPress={[Function]}
            >
              6
            </Key>
            <Key
              onPress={[Function]}
            >
              7
            </Key>
            <Key
              onPress={[Function]}
            >
              8
            </Key>
            <Key
              onPress={[Function]}
            >
              9
            </Key>
            <Key
              disabled={true}
              noBorder={true}
            />
            <Key
              onPress={[Function]}
            >
              0
            </Key>
            <Key
              onLongPress={[Function]}
              onPress={[Function]}
              pointerEvents="none"
            >
              <Icon
                allowFontScaling={false}
                color="#66688F"
                name="ios-backspace"
                size={24}
              />
            </Key>
          </Styled(View)>,
        ],
        "justify": "flex-start",
        "marginBottom": 40,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "children": <withAnimatable(View)
              delay={0}
              direction="normal"
              iterationCount={1}
              iterationDelay={0}
              onAnimationBegin={[Function]}
              onAnimationEnd={[Function]}
              onTransitionBegin={[Function]}
              onTransitionEnd={[Function]}
              useNativeDriver={false}
            >
              <Styled(View)
                align="center"
                justify="space-between"
                width={264}
              >
                <PinDigit
                  currentState={Array []}
                  digit={0}
                />
                <PinDigit
                  currentState={Array []}
                  digit={1}
                />
                <PinDigit
                  currentState={Array []}
                  digit={2}
                />
                <PinDigit
                  currentState={Array []}
                  digit={3}
                />
                <PinDigit
                  currentState={Array []}
                  digit={4}
                />
                <PinDigit
                  currentState={Array []}
                  digit={5}
                />
              </Styled(View)>
            </withAnimatable(View)>,
            "height": 180,
            "justify": "center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animation": undefined,
              "children": <Styled(View)
                align="center"
                justify="space-between"
                width={264}
              >
                <PinDigit
                  currentState={Array []}
                  digit={0}
                />
                <PinDigit
                  currentState={Array []}
                  digit={1}
                />
                <PinDigit
                  currentState={Array []}
                  digit={2}
                />
                <PinDigit
                  currentState={Array []}
                  digit={3}
                />
                <PinDigit
                  currentState={Array []}
                  digit={4}
                />
                <PinDigit
                  currentState={Array []}
                  digit={5}
                />
              </Styled(View)>,
              "delay": 0,
              "direction": "normal",
              "duration": undefined,
              "easing": undefined,
              "iterationCount": 1,
              "iterationDelay": 0,
              "onAnimationBegin": [Function],
              "onAnimationEnd": [Function],
              "onTransitionBegin": [Function],
              "onTransitionEnd": [Function],
              "style": undefined,
              "transition": undefined,
              "useNativeDriver": false,
            },
            "ref": [Function],
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": Array [
                  <PinDigit
                    currentState={Array []}
                    digit={0}
                  />,
                  <PinDigit
                    currentState={Array []}
                    digit={1}
                  />,
                  <PinDigit
                    currentState={Array []}
                    digit={2}
                  />,
                  <PinDigit
                    currentState={Array []}
                    digit={3}
                  />,
                  <PinDigit
                    currentState={Array []}
                    digit={4}
                  />,
                  <PinDigit
                    currentState={Array []}
                    digit={5}
                  />,
                ],
                "justify": "space-between",
                "width": 264,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currentState": Array [],
                    "digit": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currentState": Array [],
                    "digit": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currentState": Array [],
                    "digit": 2,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currentState": Array [],
                    "digit": 3,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currentState": Array [],
                    "digit": 4,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currentState": Array [],
                    "digit": 5,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Key
                onPress={[Function]}
              >
                1
              </Key>,
              <Key
                onPress={[Function]}
              >
                2
              </Key>,
              <Key
                onPress={[Function]}
              >
                3
              </Key>,
              <Key
                onPress={[Function]}
              >
                4
              </Key>,
              <Key
                onPress={[Function]}
              >
                5
              </Key>,
              <Key
                onPress={[Function]}
              >
                6
              </Key>,
              <Key
                onPress={[Function]}
              >
                7
              </Key>,
              <Key
                onPress={[Function]}
              >
                8
              </Key>,
              <Key
                onPress={[Function]}
              >
                9
              </Key>,
              <Key
                disabled={true}
                noBorder={true}
              />,
              <Key
                onPress={[Function]}
              >
                0
              </Key>,
              <Key
                onLongPress={[Function]}
                onPress={[Function]}
                pointerEvents="none"
              >
                <Icon
                  allowFontScaling={false}
                  color="#66688F"
                  name="ios-backspace"
                  size={24}
                />
              </Key>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "1",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "1",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "2",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "2",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "3",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "3",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "4",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "4",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "5",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "5",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "6",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "6",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "7",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "7",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "8",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "8",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "9",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "9",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": true,
                "noBorder": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "0",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "0",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Icon
                  allowFontScaling={false}
                  color="#66688F"
                  name="ios-backspace"
                  size={24}
                />,
                "onLongPress": [Function],
                "onPress": [Function],
                "pointerEvents": "none",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "allowFontScaling": false,
                  "color": "#66688F",
                  "name": "ios-backspace",
                  "size": 24,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
